node/impl/full/eth.go

```diff
@@ -2135,3 +2141,93 @@ func (g gasRewardSorter) Swap(i, j int) {
 func (g gasRewardSorter) Less(i, j int) bool {
 	return g[i].premium.Int.Cmp(g[j].premium.Int) == -1
 }
+
+func (a *EthModule) EthGetBlockReceipts(ctx context.Context, blockParam ethtypes.EthBlockNumberOrHash) ([]*api.EthTxReceipt, error) {
+	ts, err := getTipsetByEthBlockNumberOrHash(ctx, a.Chain, blockParam)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get tipset: %w", err)
+	}
+
+	// Execute the tipset to get the receipts, messages, and events
+	st, msgs, receipts, err := executeTipset(ctx, ts, a.Chain, a.StateAPI)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to execute tipset: %w", err)
+	}
+
+	// Load the state tree
+	stateTree, err := a.StateManager.StateTree(st)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to load state tree: %w", err)
+	}
+
+	// Get the parent tipset CID
+	parentTsCid, err := ts.Parents().Cid()
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get parent tipset cid: %w", err)
+	}
+
+	// Load executed messages and extract events
+	executedMsgs, err := a.EthEventHandler.EventFilterManager.LoadExecutedMessages(ctx, ts, ts)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to load executed messages: %w", err)
+	}
+
+	// Convert executed messages to events
+	events := make([]*filter.CollectedEvent, 0, len(executedMsgs))
+	for _, em := range executedMsgs {
+		for _, ev := range em.Events() {
+			addr, err := address.NewIDAddress(uint64(ev.Emitter))
+			if err != nil {
+				return nil, xerrors.Errorf("failed to create ID address: %w", err)
+			}
+			events = append(events, &filter.CollectedEvent{
+				Entries:     ev.Entries,
+				EmitterAddr: addr,
+				Height:      ts.Height(),
+				TipSetKey:   ts.Key(),
+				MsgCid:      em.Message().Cid(),
+			})
+		}
+	}
+
+	// Convert raw events to Ethereum logs
+	logs, err := ethFilterLogsFromEvents(ctx, events, a.StateAPI)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to convert events to logs: %w", err)
+	}
+
+	ethReceipts := make([]*api.EthTxReceipt, 0, len(msgs))
+	for i, msg := range msgs {
+		tx, err := newEthTx(ctx, a.Chain, stateTree, ts.Height(), parentTsCid, msg.Cid(), i)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to create Eth transaction: %w", err)
+		}
+
+		receipt, err := newEthTxReceipt(ctx, tx, &api.MsgLookup{
+			Receipt: receipts[i],
+			TipSet:  ts.Key(),
+			Height:  ts.Height(),
+		}, a.ChainAPI, a.StateAPI, a.EthEventHandler)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to create Eth receipt: %w", err)
+		}
+
+		// Add logs to the receipt
+		receipt.Logs = filterLogsByTxHash(logs, receipt.TransactionHash)
+
+		ethReceipts = append(ethReceipts, &receipt)
+	}
+
```

Comment:
As a sanity check, could you keep track of how many `receipt.Logs` end up being filtered and then comparing it to `len(logs)` just to make sure we got the same number—if we didn't then there's a problem and we should treat that as an error (which would need further investigation if someone ever encountered it).